modules/Kconfig.mcuboot

```diff
@@ -25,3 +25,9 @@ config INIT_ARCH_HW_AT_BOOT
 	  The image will be booted directly by an unknown image which might not
 	  leave the system in a clean state, so it is necessary to perform
 	  architecture specific hardware initialization.
+
+config NORDIC_QSPI_NOR
```

Comment:
I'm on the fence about these. On one hand I want to make it as easy as possible to enable external flash on our DKs. On the other hand I don't want to hide details on what is needed for using a different type of external flash for customers not using our DKs.

Also note that when the size sharing across regions is in place, if we keep these than it will be possible to fully automatically configure the mcuboot child image from the parent image.

Reply:
Just an idea, maybe @mbolivar-nordic has a better one.

Would this be a candidate for a chosen property ?
That way any extarnal flash can in principle be used, and we just rely on the chosen property.
User could then do:
```
chosen {
    ncs,dfu-flash = &some_ext_flash;
}
```
Maybe there is already an appropriate chosen to use, but I didn't spot one.
https://docs.zephyrproject.org/latest/reference/devicetree/api.html#id2 

And only if that chosen is available, then we enable the PM_EXTERNAL_FLASH... settings.
Also all size and address information should be possible to retrieve from devicetree and Kconfig could be reduced into:
Divide flash in `1` or `2` partitions (we could probably even have the partitions in dts.)

Reply:
> Would this be a candidate for a chosen property ?

What are the use cases here? The commit logs and PR description are a bit thin on details.

Reply:
Discussed offline, chosen node accepted.